File: src/test/java/com/winterhate/strategy/handlers/HandlerSecond.java
package com.winterhate.strategy.handlers;

import lombok.NonNull;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class HandlerSecond extends HandlerStrategy {
    @Override
    public @NonNull Handlers handles() {
        return Handlers.SECOND;
    }

    @Override
    public void apply(HandlerDto handlerDto) {
        log.info("Handled by HandlerSecond: {}", handlerDto);
    }
}

File: src/test/kotlin/com/winterhate/strategy/handlers/HandlerSecond.kt
package com.winterhate.strategy.handlers

import org.slf4j.Logger
import org.springframework.stereotype.Component

@Component
class HandlerSecond(private val log: Logger) : HandlerStrategy() {

    override fun handles(): Handlers = Handlers.SECOND

    override fun apply(data: HandlerDto) {
        log.info("Handled by HandlerSecond: {}", data)
    }

}

